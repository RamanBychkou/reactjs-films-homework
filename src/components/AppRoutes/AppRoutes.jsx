import React from 'react';
import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import MovieListContainer from '../../pages/MovieListContainer/index';
import MovieDetailsPage from '../../pages/MovieDetailsPage/index';
import ErrorBoundary from '../../pages/ErrorBoundary/index';

const AppRoutes = () => (
  <Switch>
    <ErrorBoundary query={{ url: '404', search: false }}>
      <Route
        exact
        path="/"
        render={() => <MovieListContainer query={{ url: '/', search: false }} />}
      />
      <Route
        path="/list/:query"
        render={props => (
          <MovieListContainer query={{ url: props.match.params.query, search: false }} />
        )}
      />

      <Route
        path="/genre/:id"
        render={props => (
          <MovieListContainer query={{ url: props.match.params.id, search: false }} />
        )}
      />
      <Route
        path="/films/:filmsName"
        render={props => (
          <MovieDetailsPage query={{ url: props.match.params.filmsName, search: false }} />
        )}
      />
      <Route
        path="/search/:query"
        render={props => (
          <MovieListContainer query={{ url: props.match.params.query, search: true }} />
        )}
      />
      <Route
        exact
        path="/404"
        render={() => (
          <MovieListContainer query={{ url: '/404', search: false }} error="Nothing Found" />
        )}
      />
    </ErrorBoundary>
  </Switch>
);

AppRoutes.defaultProps = {
  match: { params: {} },
};

AppRoutes.propTypes = {
  match: PropTypes.objectOf(PropTypes.any),
};
export default AppRoutes;
