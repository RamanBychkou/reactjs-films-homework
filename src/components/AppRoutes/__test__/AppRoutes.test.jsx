import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';

import AppRoutes from '../index';

fetchMock.config.overwriteRoutes = true;

const KEY = '75331f1a740385460b25b56203149aa8';

const genres = [
  { id: 28, name: 'Action' },
  { id: 12, name: 'Adventure' },
  { id: 16, name: 'Animation' },
  { id: 35, name: 'Comedy' },
  { id: 80, name: 'Crime' },
  { id: 99, name: 'Documentary' },
  { id: 18, name: 'Drama' },
  { id: 10751, name: 'Family' },
  { id: 14, name: 'Fantasy' },
  { id: 36, name: 'History' },
  { id: 27, name: 'Horror' },
  { id: 10402, name: 'Music' },
  { id: 9648, name: 'Mystery' },
  { id: 10749, name: 'Romance' },
  { id: 878, name: 'Science Fiction' },
  { id: 10770, name: 'TV Movie' },
  { id: 53, name: 'Thriller' },
  { id: 10752, name: 'War' },
  { id: 37, name: 'Western' },
];

describe('Test routes', () => {
  const initial = {
    error: '',
    movies: {
      mainMovie: { backdrop_path: 'test' },
      currentVideo: null,
      results: [
        {
          adult: false,
          backdrop_path: '/v4yVTbbl8dE1UP2dWu5CLyaXOku.jpg',
          genre_ids: [12],
          original_title: 'Aladdin',
          overview:
            'A kindhearted street urchin named Aladdin embarks on a magical adventure after finding a lamp that releases a wisecracking genie while a power-hungry Grand Vizier vies for the same lamp that has the power to make their deepest wishes come true.',
          popularity: 630.556,
          poster_path: '/3iYQTLGoy7QnjcUYRJy4YrAgGvp.jpg',
          release_date: '2019-05-22',
          title: 'Aladdin',
          video: false,
          vote_average: 7.2,
          vote_count: 538,
          id: 333333,
        },
      ],
    },
    genres,
    mainMovie: { backdrop_path: 'test' },
    fetchGenres: jest.fn(() => ({})),
    fetchSearchResults: jest.fn(() => ({})),
    getMainMovieDetails: jest.fn(() => ({})),
    fetchListMovies: jest.fn(() => ({})),
    clearCurrentMovie: jest.fn(() => ({})),
    setTypeView: jest.fn(),
    match: { url: '' },
  };
  const mockStore = configureMockStore([thunk]);
  const store = mockStore(initial);

  beforeEach(() => {
    fetchMock.getOnce(
      `https://api.themoviedb.org/3/search/movie?api_key=${KEY}&language=en-US&query=avengers&page=1&include_adult=false`,
      initial,
    );

    fetchMock.getOnce(
      `https://api.themoviedb.org/3/movie/popular?api_key=${KEY}&language=en-US&page=1`,
      initial,
    );

    fetchMock.getOnce(
      'https://api.themoviedb.org/3/movie/420818?api_key=75331f1a740385460b25b56203149aa8&language=en-US',
      initial,
    );

    fetchMock.getOnce(
      'https://api.themoviedb.org/3/genre/movie/list?api_key=75331f1a740385460b25b56203149aa8&language=en-US',
      [{ id: 35, name: 'Drama' }],
    );

    fetchMock.getOnce(
      `https://api.themoviedb.org/3/discover/movie?api_key=${KEY}&with_genres=35`,
      initial,
    );
  });

  it('Routes - base Routes', () => {
    const container = TestRenderer.create(
      <Provider store={store}>
        <Router initialEntries={['/']} initialIndex={0}>
          <AppRoutes />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  // Test with error
  it('Routes - list Routes', () => {
    initial.error = 'Nothing Found';
    const storeWithError = mockStore(initial);
    const container = TestRenderer.create(
      <Provider store={storeWithError}>
        <Router initialEntries={['/list/coming_soon']} initialIndex={0}>
          <AppRoutes />
        </Router>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Routes - genre Routes', () => {
    const container = TestRenderer.create(
      <Provider store={store}>
        <Router initialEntries={['/genre/35']} initialIndex={0}>
          <AppRoutes />
        </Router>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Routes - films Routes', () => {
    const container = TestRenderer.create(
      <Provider store={store}>
        <Router initialEntries={['/films/420818']} initialIndex={0}>
          <AppRoutes />
        </Router>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Routes - search Routes', () => {
    const container = TestRenderer.create(
      <Provider store={store}>
        <Router initialEntries={['/search/avengers']} initialIndex={0}>
          <AppRoutes />
        </Router>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Routes - 404 Routes', () => {
    const container = TestRenderer.create(
      <Provider store={store}>
        <Router initialEntries={['/404']} initialIndex={0}>
          <AppRoutes />
        </Router>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Routes - index.html Routes', () => {
    const container = TestRenderer.create(
      <Provider store={store}>
        <Router initialEntries={['/index.html']} initialIndex={0}>
          <AppRoutes />
        </Router>
        ,
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
