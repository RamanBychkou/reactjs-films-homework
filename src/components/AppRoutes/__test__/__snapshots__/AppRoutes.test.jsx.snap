// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test routes Routes - base Routes 1`] = `
<div
  className="wrapper"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="h1"
      >
        FILMS
      </h1>
      <form
        action=""
        onSubmit={[Function]}
      >
        <input
          className="searchField"
          id="searchField"
          tabIndex="0"
          type="search"
        />
        <input
          className="btnSearch"
          onClick={[Function]}
          type="button"
          value="⚲"
        />
      </form>
    </div>
  </header>
  <main>
    <section
      className="movieList"
    >
      <div
        className="container"
      >
        <div
          className="moviesSelector"
        >
          <nav>
            <ul>
              <li>
                <a
                  href="/list/trending"
                  onClick={[Function]}
                >
                  <button
                    href="#"
                    type="button"
                  >
                    Trending
                  </button>
                </a>
              </li>
              <li>
                <a
                  href="/list/top_rated"
                  onClick={[Function]}
                >
                  <button
                    href="#"
                    type="button"
                  >
                    Top Rated
                  </button>
                </a>
              </li>
              <li>
                <a
                  href="/list/coming_soon"
                  onClick={[Function]}
                >
                  <button
                    href="#"
                    type="button"
                  >
                    Coming Soon
                  </button>
                </a>
              </li>
              <li>
                <select
                  id=""
                  name="genre"
                  onChange={[Function]}
                >
                  <option
                    value=""
                  >
                    Genre
                  </option>
                  <option>
                    Action
                  </option>
                  <option>
                    Adventure
                  </option>
                  <option>
                    Animation
                  </option>
                  <option>
                    Comedy
                  </option>
                  <option>
                    Crime
                  </option>
                  <option>
                    Documentary
                  </option>
                  <option>
                    Drama
                  </option>
                  <option>
                    Family
                  </option>
                  <option>
                    Fantasy
                  </option>
                  <option>
                    History
                  </option>
                  <option>
                    Horror
                  </option>
                  <option>
                    Music
                  </option>
                  <option>
                    Mystery
                  </option>
                  <option>
                    Romance
                  </option>
                  <option>
                    Science Fiction
                  </option>
                  <option>
                    TV Movie
                  </option>
                  <option>
                    Thriller
                  </option>
                  <option>
                    War
                  </option>
                  <option>
                    Western
                  </option>
                </select>
              </li>
            </ul>
          </nav>
          <div
            className="typeView"
          >
            <button
              id="cards"
              onClick={[Function]}
              type="button"
            >
              ⚏
            </button>
            <button
              id="lines"
              onClick={[Function]}
              type="button"
            >
              ⚌
            </button>
          </div>
        </div>
        <div
          className="moviesWrapper"
        >
          <article
            className="movieContainer cards"
            onMouseLeave={[Function]}
          >
            <figure
              className="figcaptionBlock"
            >
              <div>
                <img
                  alt="Aladdin"
                  src="https://image.tmdb.org/t/p/w300/3iYQTLGoy7QnjcUYRJy4YrAgGvp.jpg"
                />
              </div>
              <figcaption>
                <div
                  className="movieTitle"
                >
                  <button
                    href="#"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    tabIndex="0"
                    type="button"
                  >
                    <a
                      href="/films/333333"
                      id={333333}
                      onClick={[Function]}
                    >
                      Aladdin
                    </a>
                  </button>
                  <span>
                    7.2
                  </span>
                </div>
                <div
                  className="movieGenres"
                >
                  <ul>
                    <li>
                      <button
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        type="button"
                      >
                        Adventure
                      </button>
                    </li>
                  </ul>
                </div>
              </figcaption>
            </figure>
            <div
              className="hoverMovie"
            >
              <div
                className="hoverControlBlock"
              >
                <label
                  htmlFor="playTrailer"
                  id={333333}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  <input
                    className="playTrailer"
                    id={333333}
                    name="playTrailer"
                    type="button"
                  />
                  Watch Now
                </label>
                <button
                  className="showOverwie"
                  id="showOverwie"
                  onClick={[Function]}
                  type="button"
                >
                  View Info
                </button>
              </div>
              <div
                className="hoverOverwie hoverOverwieNone"
              >
                <button
                  className="hoverClose"
                  id="closeHover"
                  onClick={[Function]}
                  type="button"
                >
                  ╳
                </button>
                <div
                  className="hoverOverwieTitle"
                >
                  <div
                    className="hoverOverwieTitleItem"
                  >
                    <h3>
                      Aladdin
                    </h3>
                    <ul>
                      <li>
                        <button
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          type="button"
                        >
                          Adventure
                        </button>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="hoverOverwieTitleItem"
                  >
                    <p>
                      7.2
                    </p>
                  </div>
                </div>
                <p>
                  A kindhearted street urchin named Aladdin embarks on a magical adventure after finding a lamp that releases a wisecracking genie while a power-hungry Grand Vizier vies for the same lamp that has the power to make their deepest wishes come true.
                </p>
                <button
                  id={333333}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  Watch Now
                </button>
              </div>
            </div>
          </article>
        </div>
      </div>
    </section>
  </main>
  <footer>
    <div
      className="container"
    >
      <h2>
        FILMS
      </h2>
      <p>
        Copyright © 2019 FILMS. RAMAN BYCHKOU
      </p>
    </div>
  </footer>
  <main>
    <div
      className="preloader"
    >
      <p>
        App is loading
      </p>
    </div>
  </main>
</div>
`;

exports[`Test routes Routes - films Routes 1`] = `
Array [
  <div>
    <header
      className="header"
    >
      <div
        className="container"
      >
        <h1
          className="h1"
        >
          FILMS
        </h1>
        <form
          action=""
          onSubmit={[Function]}
        >
          <input
            className="searchField"
            id="searchField"
            tabIndex="0"
            type="search"
          />
          <input
            className="btnSearch"
            onClick={[Function]}
            type="button"
            value="⚲"
          />
        </form>
      </div>
    </header>
    <div
      className="container"
    >
      <div
        className="title"
      >
        <h1>
          Nothing found
        </h1>
      </div>
      <div
        className="moviesSelector"
      >
        <nav>
          <ul>
            <li>
              <a
                href="/list/trending"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Trending
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/top_rated"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Top Rated
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/coming_soon"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Coming Soon
                </button>
              </a>
            </li>
            <li>
              <select
                id=""
                name="genre"
                onChange={[Function]}
              >
                <option
                  value=""
                >
                  Genre
                </option>
                <option>
                  Action
                </option>
                <option>
                  Adventure
                </option>
                <option>
                  Animation
                </option>
                <option>
                  Comedy
                </option>
                <option>
                  Crime
                </option>
                <option>
                  Documentary
                </option>
                <option>
                  Drama
                </option>
                <option>
                  Family
                </option>
                <option>
                  Fantasy
                </option>
                <option>
                  History
                </option>
                <option>
                  Horror
                </option>
                <option>
                  Music
                </option>
                <option>
                  Mystery
                </option>
                <option>
                  Romance
                </option>
                <option>
                  Science Fiction
                </option>
                <option>
                  TV Movie
                </option>
                <option>
                  Thriller
                </option>
                <option>
                  War
                </option>
                <option>
                  Western
                </option>
              </select>
            </li>
          </ul>
        </nav>
        <div
          className="typeView"
        >
          <button
            id="cards"
            onClick={[Function]}
            type="button"
          >
            ⚏
          </button>
          <button
            id="lines"
            onClick={[Function]}
            type="button"
          >
            ⚌
          </button>
        </div>
      </div>
      <img
        alt="404 Not Found"
        src="https://cdn.shopify.com/s/files/1/1164/8172/products/funny-insomnia-mug-error-404-am-sleep-not-found-11oz-black-coffee-mugs_175_1024x1024.jpg?v=1537210909"
      />
    </div>
  </div>,
  ",",
]
`;

exports[`Test routes Routes - genre Routes 1`] = `
Array [
  <div>
    <header
      className="header"
    >
      <div
        className="container"
      >
        <h1
          className="h1"
        >
          FILMS
        </h1>
        <form
          action=""
          onSubmit={[Function]}
        >
          <input
            className="searchField"
            id="searchField"
            tabIndex="0"
            type="search"
          />
          <input
            className="btnSearch"
            onClick={[Function]}
            type="button"
            value="⚲"
          />
        </form>
      </div>
    </header>
    <div
      className="container"
    >
      <div
        className="title"
      >
        <h1>
          Nothing found
        </h1>
      </div>
      <div
        className="moviesSelector"
      >
        <nav>
          <ul>
            <li>
              <a
                href="/list/trending"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Trending
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/top_rated"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Top Rated
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/coming_soon"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Coming Soon
                </button>
              </a>
            </li>
            <li>
              <select
                id=""
                name="genre"
                onChange={[Function]}
              >
                <option
                  value=""
                >
                  Genre
                </option>
                <option>
                  Action
                </option>
                <option>
                  Adventure
                </option>
                <option>
                  Animation
                </option>
                <option>
                  Comedy
                </option>
                <option>
                  Crime
                </option>
                <option>
                  Documentary
                </option>
                <option>
                  Drama
                </option>
                <option>
                  Family
                </option>
                <option>
                  Fantasy
                </option>
                <option>
                  History
                </option>
                <option>
                  Horror
                </option>
                <option>
                  Music
                </option>
                <option>
                  Mystery
                </option>
                <option>
                  Romance
                </option>
                <option>
                  Science Fiction
                </option>
                <option>
                  TV Movie
                </option>
                <option>
                  Thriller
                </option>
                <option>
                  War
                </option>
                <option>
                  Western
                </option>
              </select>
            </li>
          </ul>
        </nav>
        <div
          className="typeView"
        >
          <button
            id="cards"
            onClick={[Function]}
            type="button"
          >
            ⚏
          </button>
          <button
            id="lines"
            onClick={[Function]}
            type="button"
          >
            ⚌
          </button>
        </div>
      </div>
      <img
        alt="404 Not Found"
        src="https://cdn.shopify.com/s/files/1/1164/8172/products/funny-insomnia-mug-error-404-am-sleep-not-found-11oz-black-coffee-mugs_175_1024x1024.jpg?v=1537210909"
      />
    </div>
  </div>,
  ",",
]
`;

exports[`Test routes Routes - index.html Routes 1`] = `","`;

exports[`Test routes Routes - list Routes 1`] = `
Array [
  <div>
    <header
      className="header"
    >
      <div
        className="container"
      >
        <h1
          className="h1"
        >
          FILMS
        </h1>
        <form
          action=""
          onSubmit={[Function]}
        >
          <input
            className="searchField"
            id="searchField"
            tabIndex="0"
            type="search"
          />
          <input
            className="btnSearch"
            onClick={[Function]}
            type="button"
            value="⚲"
          />
        </form>
      </div>
    </header>
    <div
      className="container"
    >
      <div
        className="title"
      >
        <h1>
          Nothing found
        </h1>
      </div>
      <div
        className="moviesSelector"
      >
        <nav>
          <ul>
            <li>
              <a
                href="/list/trending"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Trending
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/top_rated"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Top Rated
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/coming_soon"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Coming Soon
                </button>
              </a>
            </li>
            <li>
              <select
                id=""
                name="genre"
                onChange={[Function]}
              >
                <option
                  value=""
                >
                  Genre
                </option>
                <option>
                  Action
                </option>
                <option>
                  Adventure
                </option>
                <option>
                  Animation
                </option>
                <option>
                  Comedy
                </option>
                <option>
                  Crime
                </option>
                <option>
                  Documentary
                </option>
                <option>
                  Drama
                </option>
                <option>
                  Family
                </option>
                <option>
                  Fantasy
                </option>
                <option>
                  History
                </option>
                <option>
                  Horror
                </option>
                <option>
                  Music
                </option>
                <option>
                  Mystery
                </option>
                <option>
                  Romance
                </option>
                <option>
                  Science Fiction
                </option>
                <option>
                  TV Movie
                </option>
                <option>
                  Thriller
                </option>
                <option>
                  War
                </option>
                <option>
                  Western
                </option>
              </select>
            </li>
          </ul>
        </nav>
        <div
          className="typeView"
        >
          <button
            id="cards"
            onClick={[Function]}
            type="button"
          >
            ⚏
          </button>
          <button
            id="lines"
            onClick={[Function]}
            type="button"
          >
            ⚌
          </button>
        </div>
      </div>
      <img
        alt="404 Not Found"
        src="https://cdn.shopify.com/s/files/1/1164/8172/products/funny-insomnia-mug-error-404-am-sleep-not-found-11oz-black-coffee-mugs_175_1024x1024.jpg?v=1537210909"
      />
    </div>
  </div>,
  ",",
]
`;

exports[`Test routes Routes - search Routes 1`] = `
Array [
  <div>
    <header
      className="header"
    >
      <div
        className="container"
      >
        <h1
          className="h1"
        >
          FILMS
        </h1>
        <form
          action=""
          onSubmit={[Function]}
        >
          <input
            className="searchField"
            id="searchField"
            tabIndex="0"
            type="search"
          />
          <input
            className="btnSearch"
            onClick={[Function]}
            type="button"
            value="⚲"
          />
        </form>
      </div>
    </header>
    <div
      className="container"
    >
      <div
        className="title"
      >
        <h1>
          Nothing found
        </h1>
      </div>
      <div
        className="moviesSelector"
      >
        <nav>
          <ul>
            <li>
              <a
                href="/list/trending"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Trending
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/top_rated"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Top Rated
                </button>
              </a>
            </li>
            <li>
              <a
                href="/list/coming_soon"
                onClick={[Function]}
              >
                <button
                  href="#"
                  type="button"
                >
                  Coming Soon
                </button>
              </a>
            </li>
            <li>
              <select
                id=""
                name="genre"
                onChange={[Function]}
              >
                <option
                  value=""
                >
                  Genre
                </option>
                <option>
                  Action
                </option>
                <option>
                  Adventure
                </option>
                <option>
                  Animation
                </option>
                <option>
                  Comedy
                </option>
                <option>
                  Crime
                </option>
                <option>
                  Documentary
                </option>
                <option>
                  Drama
                </option>
                <option>
                  Family
                </option>
                <option>
                  Fantasy
                </option>
                <option>
                  History
                </option>
                <option>
                  Horror
                </option>
                <option>
                  Music
                </option>
                <option>
                  Mystery
                </option>
                <option>
                  Romance
                </option>
                <option>
                  Science Fiction
                </option>
                <option>
                  TV Movie
                </option>
                <option>
                  Thriller
                </option>
                <option>
                  War
                </option>
                <option>
                  Western
                </option>
              </select>
            </li>
          </ul>
        </nav>
        <div
          className="typeView"
        >
          <button
            id="cards"
            onClick={[Function]}
            type="button"
          >
            ⚏
          </button>
          <button
            id="lines"
            onClick={[Function]}
            type="button"
          >
            ⚌
          </button>
        </div>
      </div>
      <img
        alt="404 Not Found"
        src="https://cdn.shopify.com/s/files/1/1164/8172/products/funny-insomnia-mug-error-404-am-sleep-not-found-11oz-black-coffee-mugs_175_1024x1024.jpg?v=1537210909"
      />
    </div>
  </div>,
  ",",
]
`;
